Update docs to reflect discussion (#24)
bcc-code/bcc-media-adr: 02-analytics/technical.md

### Data

#### Signed in users
`userId`: Accessible from GQL under `me` endpoint

`traits`

| Data | Name | Comments |
|------|------|----------|
| Unique user ID | `id` | Same as the `userId` |
| Users exact age | `age` | Currently not used. Need to figure out how to guarantee anonymity |
| Age Group | `ageGroup` | See Age group List below |
| Country | `country` | Two letter country code |

> | Church ID | `church` | Numerical church ID |

Author: ```suggestion
| Church ID | `church` | Numerical church ID as STRING |
```

> | Gender | `gender` | M or F |

Reviewer: can it be "male" or "female"? to match the idtoken

#### Age groups

Please use the exact strings below:

 * UNKNOWN
 * < 10
 * 10 - 12
 * 13 - 18

>  * 19 - 26

Reviewer: ```suggestion
 * 19 - 25
```

still a conflict here with 19 - 26 and 26 - 36

 * 26 - 36
 * 37 - 50
 * 51 - 65

>  * 65+

Reviewer 2: 27 - 36 ?
51 - 64 ?

Reviewer: 19-25 is probably better
26-36

51-64

>  * UNKNOWN
>  * < 10
>  * 10 - 12
>  * 13 - 18
>  * 19 - 26
>  * 26 - 36
>  * 37 - 50
>  * 51 - 65
>  * 65+

Author: ```suggestion
 * UNKNOWN
 * < 10
 * 10 - 12
 * 13 - 18
 * 19 - 25
 * 26 - 36
 * 37 - 50
 * 51 - 65
 * 65+
```

| Data         | Name        | Comments                                       |
|--------------|-------------|------------------------------------------------|
| ID | Id | Code of the page if generated or from the list below. String |
| Title | title | String, native language |
| Additional info | meta | JSON. Currently unused, but fields can be added as needed |

### Page/Screen IDs

* about
* audience
* calendar
* liveStream
* login
* profileEdit
* profile
* search

> * settings

Author: I thin a general `settings` is more flexible.

> * settings

Reviewer: ok

> * settings

Reviewer: settings with extra property "setting": appLanguage, audioLanguage, subtitleLanguage

> * settings

Reviewer 2: On web this might be slightly different as we have more space and likely show all options at the same time.

Perhaps settings/profile won't even be a separate page, just a modal, so I might need another event type for that or?

> * settings

Author: I would say that it could still count as a page, and just send maybe "all" for the event type?

> * settings

Reviewer: I agree that it can be a page even though it's just a dropdown or modal

> * settings

Author: I updated the text and are resolving this. Please open a new convo if there is more.

| Data         | Name        | Comments                                       |
|--------------|-------------|------------------------------------------------|
| ID | Id | Code of the page if generated or from the list below. String |
| Title | title | String, native language |
| Additional info | meta | JSON. Currently unused, but fields can be added as needed |

### Page/Screen IDs

* about
* calendar
* livestream
* login
* profileEdit
* profile
* search
* settings
* support
* faq
* episode

> ## Category item click (category_item_click)

Reviewer: this isnt a thing anymore

| Data         | Name        | Comments                                       |
|--------------|-------------|------------------------------------------------|
| ID | Id | Code of the page if generated or from the list below. String |
| Title | title | String, native language |
| Additional info | meta | JSON. See below for `settings` example, but fields can be added as needed |

### Page/Screen IDs

* about
* calendar
* livestream
* login
* profileEdit
* profile
* search
* settings
* support
* faq

> * episode

Reviewer 2: What do we use for the dynamic pages?

Author: The page code

Author: Is it important in terms of analytics? I just wanna know what page was visited, do I relly care about it being dynamic or not?

Reviewer: yes, the generic pages like settings etc create noise when analyzing performance of those pages. Being able to filter them out by checking existence of a `pageCode` prop helps. 
"home" and "search" should include the extra "pageCode" to signal the dynamic page used.

Author: Ok, agree!

Reviewer: added pageCode then

| Element Type     | elementType     | episode, series, ...                                                                       |
| Element ID       | elementId       | id of the clicked element                                                                  |

> ## Section click (section_clicked)

Reviewer: **sectionType**: Kan vi bruke __typename fra graphql? DefaultSection, DefaultGridSection etc etc etc.
**elementType**: Kan vi bruke item.__typename fra graphql? Episode, Show, etc etc.

Reviewer: Episode
```json
    "properties": {
        "elementId": "1618",
        "elementPosition": 2,
        "elementType": "Episode",
        "sectionId": "1",
        "sectionName": "Latest episodes",
        "sectionPosition": 5,
        "sectionType": "DefaultSection"
    },
```

Show
```json
    "properties": {
        "elementId": "93",
        "elementPosition": 0,
        "elementType": "Show",
        "sectionId": "55",
        "sectionName": "Våre høytidsstevner",
        "sectionPosition": 11,
        "sectionType": "DefaultSection"
    },
```

Author: I think that sounds sane.

> ## Section click (section_clicked)

Reviewer: I would add `elementName` too.

| See All          | seeAll          | true if the tapped element was "See All"      |
| Page Name        | pageName        | same ID as for page/screen tracking           |

> ## Audio Only (video_toggled)

Author: ```suggestion
## Audio Only (audioonly_clicked)
```

> | See All          | seeAll          | true if the tapped element was "See All"      |

Reviewer: ```suggestion
```

Does not exist anymore

> | Page Name        | pageName        | same ID as for page/screen tracking           |

Reviewer: ```suggestion
| Page Code        | pageCode        | same ID as for page/screen tracking           |
```

| Data        | Name             | Comments                     |
|-------------|------------------|------------------------------|

> | Event ID    | event            | Hardcoded: `video_toggled`    |

Author: ```suggestion
| Event ID    | event            | Hardcoded: `audioonly_clicked`    |
```

| Data        | Name             | Comments                     |
|-------------|------------------|------------------------------|
| Event ID    | event            | Hardcoded: `audioonly_clicked`    |

> | Video state | videoStateTarget | Video state *after* tap      |

Reviewer 2: What is video state? 
"video" | "audio_only" ?

Reviewer: ```suggestion
| Audio only enabled | audioOnly | Audio only state *after* tap. true or false      |
```

| Data          | Name         | Comments                                      |
|---------------|--------------|-----------------------------------------------|
| Event ID      | event        | Hardcoded: `calendarentry_clicked`                 |

> | Page Name     | pageName     | same ID as for page/screen tracking           |

Reviewer 2: Stick to a convention? `pageCode` on section_clicked

Reviewer: ```suggestion
| Page Code     | pageCode     | same code as for page/screen tracking           |
```

| Data               | Name                        | Commnets                 |
|--------------------|-----------------------------|--------------------------|
| Event ID           | eventID                     | `notification_received`   |

> | Notification id    | notificationId              | for tracking purposes    |

Reviewer: ```suggestion
| Notification id    | notificationId              | for tracking purposes    |
| action         | action         | `deep_link`, `clear_cache`, null  |
| deeplink         | string         |   if action is `deep_link`  |
```
